File: Part1_강의자료/Ch03_배열/문제별코드/10431_줄세우기/10431.py
P = int(input())

for T in range(1, P + 1):
    h = list(map(int, input().split())) [1:]
    cnt = 0

    for i in range(20):
        for j in range(i):
            if h[j] > h[i]:
                myH = h[i]
                for k in range(i, j, -1):
                    h[k] = h[k - 1]
                    cnt += 1
                h[j] = myH
                break

    print(T, cnt)

File: Part1_강의자료/Ch03_배열/문제별코드/10431_줄세우기/10431.cpp
#include <stdio.h>

int P, T;
int h[20];

int main()
{
    scanf("%d", &P);
    while (P--)
    {
        scanf("%d", &T);
        for(int i = 0; i < 20; i++)
            scanf("%d", h + i);

        int cnt = 0;
        for(int i = 0; i < 20; i++)
        {
            for(int j = 0; j < i; j++)
            {
                if (h[j] > h[i])
                {
                    int myH = h[i];
                    for(int k = i; k > j; k--)
                    {
                        h[k] = h[k - 1];
                        cnt++;
                    }
                    h[j] = myH;
                    break;
                }
            }
        }

        printf("%d %d\n", T, cnt);
    }
    return 0;
}
